pkg/cmd/roachtest/tests/failure_injection.go

```diff
@@ -324,58 +343,312 @@ var latencyTest = func(c cluster.Cluster) failureSmokeTest {
+					l.Printf("Stalled nodes: %d, Unaffected nodes: %d, Stalled validation node: %d, Unaffected validation node: %d", stalledNodeGroup, unaffectedNodeGroup, stalledNode, unaffectedNode)
+					res, err := getRWBytesOverTime(ctx, l, c, f.(*failures.CGroupDiskStaller), stalledNode, unaffectedNode)
+					if err != nil {
+						return err
+					}
+					l.Printf("ReadBytes and WriteBytes over last 30 seconds:%+v", res)
+
+					return assertRWBytes(ctx, l, res, stallReads, stallWrites)
+				},
+				validateRecover: func(ctx context.Context, l *logger.Logger, c cluster.Cluster, f failures.FailureMode) error {
+					// Wait for replication since the stalled node may have just restarted.
+					// TODO(darryl): The failure mode itself should do this in WaitForFailureToRecover.
+					// It should also wait for replicas to rebalance, although this test is not large
+					// enough for that to matter.
+					db := c.Conn(ctx, l, stalledNode[0])
+					defer db.Close()
+					if err := roachtestutil.WaitForReplication(ctx, l, db, 3 /* replicationFactor */, roachtestutil.AtLeastReplicationFactor); err != nil {
+						return err
+					}
+					res, err := getRWBytesOverTime(ctx, l, c, f.(*failures.CGroupDiskStaller), stalledNode, unaffectedNode)
+					if err != nil {
+						return err
+					}
+					l.Printf("ReadBytes and WriteBytes over last 30 seconds:%+v", res)
+
+					// The cluster should be fully recovered from the stalls, so assert that
+					// reads and writes are not stalled on both the stalled and unaffected nodes.
+					return assertRWBytes(ctx, l, res, false /*readsStalled*/, false /*writesStalled*/)
+				},
+				workload: func(ctx context.Context, c cluster.Cluster) error {
+					cmd := roachtestutil.NewCommand("./cockroach workload run kv").
+						Arg("{pgurl%s}", c.CRDBNodes()).
+						// Because disk stalls should take down a node, we need to tolerate errors or
+						// the workload will error out failing to connect to stalledNode.
+						Arg("--tolerate-errors").
+						Arg("--sequential").
+						Flag("cycle-length", 1000).
+						Flag("read-percent", 50).
+						String()
+					return c.RunE(ctx, option.WithNodes(c.WorkloadNode()), cmd)
+				},
+			})
+		}
+	}
+
+	return tests
+}
+
+var dmsetupDiskStallTest = func(c cluster.Cluster) failureSmokeTest {
+	rng, _ := randutil.NewPseudoRand()
+	// SeededRandGroups only returns an error if the requested size is larger than the
+	// number of nodes, so we can safely ignore the error.
+	groups, _ := c.CRDBNodes().SeededRandGroups(rng, 2 /* numGroups */)
+	stalledNodeGroup := groups[0]
+	unaffectedNodeGroup := groups[1]
+	// These are the nodes that we will run validation on.
+	stalledNode := stalledNodeGroup.SeededRandNode(rng)
+	unaffectedNode := unaffectedNodeGroup.SeededRandNode(rng)
+
+	ableToCreateFile := func(ctx context.Context, l *logger.Logger, c cluster.Cluster, node option.NodeListOption) bool {
```

Comment:
Nit: `ableToCreateFile` -> `touchFile`. Since we can't tell exactly why this func failed, e.g., ssh timeout, we should use a "weaker", i.e., less precise, name. Otherwise, the name could mislead the author/reader into thinking that the `false` result is caused by something else entirely, e.g., file permissions.